### PR TITLE
Revert "[SPARK-28093][SQL] Fix TRIM/LTRIM/RTRIM function parameter order issue"

### PR DESCRIPTION
**Pull request by Ngone51**
### What changes were proposed in this pull request?

This reverts commit bef5d9d6c348e390f99b2cd781a2471d635e55f8.

### Why are the changes needed?

Revert it according to https://github.com/apache/spark/pull/24902#issuecomment-584511167.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

Pass Jenkins.

### CODE REVIEW

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/StringExpressionsSuite.scala [hunk 1]
```diff
@@ -559,9 +559,6 @@ class StringExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     // scalastyle:on
     checkEvaluation(StringTrim(Literal("a"), Literal.create(null, StringType)), null)
     checkEvaluation(StringTrim(Literal.create(null, StringType), Literal("a")), null)
-
-    checkEvaluation(StringTrim(Literal("yxTomxx"), Literal("xyz")), "Tom")
-    checkEvaluation(StringTrim(Literal("xxxbarxxx"), Literal("x")), "bar")
```
Comment (cloud-fan) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/StringExpressionsSuite.scala]: can we still keep the tests with the actual parameter order?

File: sql/core/src/test/resources/sql-tests/inputs/string-functions.sql [hunk 2]
```diff
@@ -38,13 +38,3 @@ SELECT substring('Spark SQL', 5, 1);
 SELECT substring('Spark SQL' from 5);
 SELECT substring('Spark SQL' from -3);
 SELECT substring('Spark SQL' from 5 for 1);
-
--- trim/ltrim/rtrim
-SELECT trim('yxTomxx', 'xyz'), trim(BOTH 'xyz' FROM 'yxTomxx'), trim('xyz' FROM 'yxTomxx');
```
Comment (cloud-fan) [inline sql/core/src/test/resources/sql-tests/inputs/string-functions.sql]: ditto

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/StringExpressionsSuite.scala (hunk 1 above)
Comment (Ngone51) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/StringExpressionsSuite.scala]: sure.

Comment (cloud-fan) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/StringExpressionsSuite.scala]: We need to update the test

**Review by HyukjinKwon — APPROVED** [review]

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/StringExpressionsSuite.scala (hunk 1 above)
Comment (Ngone51) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/StringExpressionsSuite.scala]: The order changed in #24902 didn't has affect on constructor `StringTrim(srcStr, trimStr)` and so test results don't change before/after #24902. So, I feel it's unnecessary to update parameter order here. And we do update parameter order in sql test(`string-expressions.sql`). WDYT?

Comment (HyukjinKwon) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/StringExpressionsSuite.scala]: What about we just revert here cleanly and add a test in a separate PR?

Comment (cloud-fan) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/StringExpressionsSuite.scala]: sounds good.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala [hunk 3]
```diff
@@ -719,7 +719,7 @@ object StringTrim {
 
     _FUNC_(TRAILING FROM str) - Removes the trailing space characters from `str`.
 
-    _FUNC_(str, trimStr) - Remove the leading and trailing `trimStr` characters from `str`.
+    _FUNC_(trimStr, str) - Remove the leading and trailing `trimStr` characters from `str`.
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala]: I'm thinking about if we should remove this line (as well as the example) and hide this API. This was definitely a mistake and it's not even consistent with the scala version in `functions.scala`. For backward compatibility, we should keep this mistake but don't expose it too publicly.

I think `trimStr FROM str` should be the recommended API at the SQL side.

Comment (cloud-fan) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala]: cc @wangyum @viirya @maropu

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Hi, All.
It's not clear to me. What is the purpose of this revert?
> For backward compatibility, we should keep this mistake but don't expose it too publicly.

**Comment by dongjoon-hyun** [comment]
@cloud-fan , @HyukjinKwon , @gatorsmile .
For this reverting, I believe we need to discuss on the mailing list first.
I'm okay that we need to choose one way or the other.
But, without explicit discussion, I'm -1 for this reverting PR. This PR should not land on `master`.

**Comment by dongjoon-hyun** [comment]
Please reopen this after we made a decision on Apache eSpark dev mail discussion.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala (hunk 3 above)
Comment (maropu) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala]: I personally like the explicit format like `trimStr FROM str` and the idea to hide the api in future releases look reasonable to me. If we fix it that way, I think its better to fix the other string functions consistently, e.g., `substring` and `overlay`.

These string functions are pretty common in dbms-like systems, so I feel throwing an exception is better than returning a wrong answer when users just copy&paste a query including these string functions from another system.

Comment (viirya) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala]: Hmm, this looks like a mistake at all. Seems no other feasible solution for this. I'm fine to remove it from doc. One question is shall we add some words in migration guide? Can we say we deprecate it and then we will make correct one in future version?

**Comment by cloud-fan** [comment]
according to the dev list discussion, we want to continue the revert.

**Comment by cloud-fan** [comment]
retest this please

**Comment by dongjoon-hyun** [comment]
Yep. +1 for reverting.

**Comment by dongjoon-hyun** [comment]
Hi, All.
Please review the following together.
- https://github.com/apache/spark/pull/27643 SPARK-30886 Warn two-parameter TRIM/LTRIM/RTRIM function usages

**Comment by maropu** [comment]
retest this please

File: sql/core/src/test/resources/sql-tests/inputs/string-functions.sql (hunk 2 above)
Comment (cloud-fan) [inline sql/core/src/test/resources/sql-tests/inputs/string-functions.sql]: We should keep the tests that use the FROM syntax.

File: sql/core/src/test/resources/sql-tests/inputs/string-functions.sql [hunk 4]
```diff
@@ -40,11 +40,11 @@ SELECT substring('Spark SQL' from -3);
 SELECT substring('Spark SQL' from 5 for 1);
 
 -- trim/ltrim/rtrim
```
Comment (cloud-fan) [inline sql/core/src/test/resources/sql-tests/inputs/string-functions.sql]: nit: only `trim`

**Review by cloud-fan — APPROVED** [review]

**Review by maropu — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master/3.0!

**Comment by Ngone51** [comment]
thanks all!!

**Pull request closed without merging**